--- libpostproc/postprocess.c.old	2017-04-23 15:01:29.522247979 +0200
+++ libpostproc/postprocess.c	2017-04-23 15:01:33.641196485 +0200
@@ -961,7 +961,7 @@
 
         if (caps & AV_CPU_FLAG_MMX)
             cpuCaps |= PP_CPU_CAPS_MMX;
-        if (caps & AV_CPU_FLAG_MMX2)
+        if (caps & AV_CPU_FLAG_MMXEXT)
             cpuCaps |= PP_CPU_CAPS_MMX2;
         if (caps & AV_CPU_FLAG_3DNOW)
             cpuCaps |= PP_CPU_CAPS_3DNOW;
